File: NetBlog.UI/NetBlog/src/app/Features/Posts/models/PostDTO.ts
import {UserShortcutDTO} from "../../Auth/models/UserShortcutDTO";
import {CommentDTO} from "../Comments/models/CommentDTO";

export interface PostDTO {
  id: string;
  title: string;
  content: string;
  dateCreated: Date;
  dateUpdated?: Date;
  createdBy: UserShortcutDTO;
  comments: CommentDTO[];
}

File: NetBlog.API/NetBlog.Common/DTO/PostDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace NetBlog.Common.DTO
{
    public class PostDTO
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public string Content { get; set; }
        public DateTime DateCreated { get; set; }
        public DateTime? DateUpdated { get; set; }
        public UserShortcutDTO CreatedBy { get; set; }
        public IEnumerable<CommentDTO> Comments { get; set; }
    }
}
